shopinvader/services/cart.py

```diff
@@ -351,20 +386,39 @@ def _get_step_from_code(self, code):
         else:
             return step
 
-    def _to_json(self, cart):
-        if not cart:
-            return {
-                "data": {},
-                "store_cache": {"cart": {}},
-                "set_session": {"cart_id": 0},
-            }
-        res = super(CartService, self)._to_json(cart)[0]
+    def _to_json_simple(self, cart):
+        """
+        Returns simple and fast items
+        :return: dict
+        """
+        cart_simple = cart.with_context(prefetch_fields=False)
+        qty = sum(
+            line.product_uom_qty
+            for line in cart_simple.order_line
+            if self._is_item(line)
+        )
+        return {"id": cart.id, "lines": {"count": qty}}
 
-        return {
-            "data": res,
-            "set_session": {"cart_id": res["id"]},
-            "store_cache": {"cart": res},
-        }
+    def _to_json(self, cart, simple=False):
+        """
+        Return cart json depending on which type of service (simple or not)
+        Modify Session response accordingly
+        :param cart:
+        :param simple:
+        :return:
+        """
+        response = shopinvader_response.get()
+        if not cart:
+            response.set_session("cart_id", 0)
+            response.set_store_cache("cart", {})
+            return {"data": {}}
+        if simple:
+            res = self._to_json_simple(cart)
```

Comment:
I think we should return a flag to say explicitely if the cart is simple or not.
simple: true/false

Indeed, on frontend side when you do a "store.cart" you may have a simple or full version (depending if the last call was an update/add or a get)

Reply:
Ok. Let's add this. Maybe after the merge on another PR ?

Reply:
yes Denis in another PR please. it makes sense to be more generic

Reply:
yes other PR will be perfect